card/mod/03_machines/lib/card/machine.rb

```diff
@@ -151,12 +151,36 @@ def run_machine joint="\n"
 
     def run_engine input_card
       return if input_card.is_a? Card::Set::Type::Pointer
+      if (cached = fetch_cache_card(input_card))
+        return cached.content
+      end
+
       input = if input_card.respond_to? :machine_input
                 input_card.machine_input
               else
                 input_card.format._render_raw
               end
-      engine(input)
+      output = engine(input)
+      cache_output_part input_card, output
+      output
+    end
+
+    def fetch_cache_card input_card, new=nil
+      new &&= { type_id: PlainTextID }
+      Card.fetch input_card.name, :machine_cache, name, new: new
```

Comment:
eg of new API idea,

this would probably be:
```
Card.fetch input_card.name, name solid_cache: { type_id: PlainTextID }
```
...and then you wouldn't need any of the save! handling.